### PR TITLE
feat: error messages when connecting to mssql

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
Added custom error messages for an incorrect database name, cleaned up some language in other tests. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2021-04-15 at 12 14 47 PM
### TEST PLAN
added unit tests

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/errors.py [hunk 1]
```diff
@@ -179,7 +179,13 @@ class SupersetErrorType(str, Enum):
         {
             "code": 1014,
             "message": _("Issue 1014 - Either the username or the password is wrong."),
-        }
+        },
+        {
+            "code": 1015,
+            "message": _(
+                "Issue 1015 - The database is " "spelled incorrectly or does not exist."
```
Comment (betodealmeida) [inline superset/errors.py:186]:
```suggestion
                "Issue 1015 - The database is spelled incorrectly or does not exist."
```

**Review by betodealmeida — CHANGES_REQUESTED** [review]
The copy in issue 1015 should be identical everywhere.

**Comment by AAfghahi** [comment]
sounds good, will update and then ping you.

**Review by betodealmeida — APPROVED** [review]
Awesome sauce!

**Pull request merged**